spark/src/main/java/org/apache/iceberg/actions/RemoveOrphanFilesAction.java

```diff
@@ -79,6 +87,20 @@
     }
   }, DataTypes.StringType);
 
+  private static final UserDefinedFunction decodeUDF = functions.udf((String fullyQualifiedPath) -> {
+    return URLDecoder.decode(fullyQualifiedPath, "UTF-8");
+  }, DataTypes.StringType);
+
+  /**
+   * Transform a file path to
+   * {@code Dataset<Row<file_path_no_scheme_authority, file_path_with_scheme_authority>>}
+   */
+  private static final UserDefinedFunction addFilePathOnlyUDF = functions.udf((String fullyQualifiedPath) -> {
+    Path path = new Path(fullyQualifiedPath);
+    // only_path, fully qualified path
+    return RowFactory.create(path.toUri().getPath(), path.toUri().toString());
+  }, DataTypes.createStructType(fileDetailStructType().fields()));
```

Comment:
Isn't this already a struct type? why are we creating a structtype from it?

Reply:
See comment here https://github.com/apache/iceberg/pull/1471/files#r490575150 for more thoughts